### PR TITLE
fix: cpu system info detection on macOS

### PR DESCRIPTION
**Pull request by ilyam8**
##### Summary

Fixes: #12291

This PR fixes detection of the following parameters on macOS:
 - CPU_LOGICAL_CPU_COUNT
 - CPU_VENDOR
 - CPU_FREQ (only for x86_64)
 - CPU_MODEL

##### Test Plan

Run the system-info.sh script on intel/m1 mac's, check the CPU info.

```
wget -q -O system-info.sh https://raw.githubusercontent.com/ilyam8/netdata/fix_macos_cpu_system_info/daemon/system-info.sh; sh system-info.sh | grep -i cpu
```

- m1 pro

```cmd
NETDATA_SYSTEM_CPU_LOGICAL_CPU_COUNT=10
NETDATA_SYSTEM_CPU_VENDOR=Apple
NETDATA_SYSTEM_CPU_MODEL=Apple M1 Pro
NETDATA_SYSTEM_CPU_FREQ=unknown
NETDATA_SYSTEM_CPU_DETECTION=nproc sysctl
```

- intel

```cmd
NETDATA_SYSTEM_CPU_LOGICAL_CPU_COUNT=8
NETDATA_SYSTEM_CPU_VENDOR=GenuineIntel
NETDATA_SYSTEM_CPU_MODEL=Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
NETDATA_SYSTEM_CPU_FREQ=2300000000
NETDATA_SYSTEM_CPU_DETECTION=sysctl
```

### CODE REVIEW

File: daemon/system-info.sh [hunk 1]
```diff
@@ -224,8 +227,15 @@ else
...
 
-  # If we have GNU uname, we can use that to get CPU info (probably).
-  if uname --version 2> /dev/null | grep -qF 'GNU coreutils'; then
+  if [ "${KERNEL_NAME}" = Darwin ]; then
+    CPU_MODEL="$(sysctl -n machdep.cpu.brand_string)"
+    if [ "${ARCHITECTURE}" = "x86_64" ]; then
+      CPU_VENDOR="$(sysctl -n machdep.cpu.vendor)"
+    else
+      CPU_VENDOR="Apple"
+    fi
```
Comment (ilyam8) [inline daemon/system-info.sh:236]:
```suggestion
    if [ "${ARCHITECTURE}" = "x86_64" ]; then
      CPU_VENDOR="$(sysctl -n machdep.cpu.vendor)"
    fi
```

Don't hardcode vendor for m1?

Comment (Ferroin) [reply daemon/system-info.sh:236]: Can we reliably get this info even in cases where getting the CPU frequency fails?

Comment (ilyam8) [reply daemon/system-info.sh:236]: What do you mean?. The current implementation uses `sysctl` utility to retrieve CPU info. If the kernel returns the value we need - we are good, if not - 🤷‍♂️.

Comment (ilyam8) [reply daemon/system-info.sh:236]: > Can we reliably get this info even in cases where getting the CPU frequency fails?

Ah, I see. Yes, our tests:
 - me checked on m1 pro.
 - @cpipilas checked on an intel mac.

**Review by thiagoftsm — APPROVED** [review]
I do not have an apple, but on Intel computer the script is given the expected result.

**Review by cpipilas — APPROVED** [review]
Works on intel mac

**Pull request merged**